### ISSUE TITLE
Download Publications - Out of Memory

### ISSUE THREAD

**Issue by EmilyHaag** [outbreak-info/outbreak.info#406]
Twitter user reported that if you "View entire library" and "Download results," the webpage will crash around ~50% complete.

I tried replicating the issue and get **Error code: out of memory** after 65% complete
Tried different browsers, clearing cache, updating browsers, and troubleshooting memory issues.

**Comment by gkarthik** [comment]
The entire library is probably too big for the browser to handle in memory now. Solution from @flaneuse to download directly from API endpoint - `https://api.outbreak.info/resources/query?fetch_all=true`

The Python code below should do the trick,

```
import requests
import time

r = requests.get("https://api.outbreak.info/resources/query?fetch_all=true")
d = r.json()
hits = d["hits"]

while "hits" in d:
    r = requests.get("https://api.outbreak.info/resources/query?fetch_all=true&scroll_id={}".format(d["_scroll_id"]))
    d = r.json()
    hits.extend(d["hits"])
    time.sleep(0.5) # To avoid request throttling
```

**Comment by flaneuse** [comment]
Yeah, it might be possible to download it in chunks but that'll take some time to do.

**Comment by nahid18** [comment]
> The entire library is probably too big for the browser to handle in memory now. Solution from @flaneuse to download directly from API endpoint - `https://api.outbreak.info/resources/query?fetch_all=true`
> 
> The Python code below should do the trick,
> 
> ```
> import requests
> import time
> 
> r = requests.get("https://api.outbreak.info/resources/query?fetch_all=true")
> d = r.json()
> hits = d["hits"]
> 
> while "hits" in d:
>     r = requests.get("https://api.outbreak.info/resources/query?fetch_all=true&scroll_id={}".format(d["_scroll_id"]))
>     d = r.json()
>     hits.extend(d["hits"])
>     time.sleep(0.5) # To avoid request throttling
> ```

This code works and thanks for this!
But what if I try to download only the litcovid ones instead of fetching all? In that case, how the api endpoint should be changed?
